lib/src/call/quality_issue.dart

```diff
@@ -0,0 +1,398 @@
+  VIQualityIssueTypeIceDisconnected,
+
+  /// Indicates that no audio is captured by the microphone.
+  VIQualityIssueTypeNoAudioSignal,
+
+  /// Indicates packet loss for last 2.5 seconds.
+  VIQualityIssueTypePacketLoss,
+
+  /// Indicates that no audio is received on a remote audio stream.
+  ///
+  /// The issue level obtained may be:
+  /// * [VIQualityIssueLevel.VIQualityIssueLevelNone] - that indicates that
+  /// audio is receiving on all remote audio streams
+  /// * [VIQualityIssueLevel.VIQualityIssueLevelCritical] - that indicates a
+  /// problem with audio receive on at least one remote audio stream
+  VIQualityIssueTypeNoAudioReceive,
+
+  /// Indicates that no video is received on a remote video stream.
+  ///
+  /// The issue level obtained may be:
+  /// * [VIQualityIssueLevel.VIQualityIssueLevelNone] - that indicates that
+  /// video is receiving on all remote video streams according to their configuration
+  /// * [VIQualityIssueLevel.VIQualityIssueLevelCritical] - that indicates a
+  /// problem with video receive on at least one remote video stream
+  VIQualityIssueTypeNoVideoReceive,
+}
+
+/// Represents send or captured a frame size.
+class FrameSize {
+  double width;
+  double height;
+
+  FrameSize({required this.width, required this.height});
+}
+
+/// Represents issues that affect call quality during a call.
+abstract class QualityIssueEvent {}
+
+/// Represents class reporting that local video is encoded by a codec
+/// different from specified in `VICallSettings.preferredVideoCodec`.
+///
+/// Issue level is `VIQualityIssueLevel.VIQualityIssueLevelCritical` if video
+/// is not sent, `VIQualityIssueLevel.VIQualityIssueLevelMajor` in case of
+/// codec mismatch or `VIQualityIssueLevel.VIQualityIssueLevelNone` if the issue
+/// is not detected.
+///
+/// Possible reasons:
+/// * The video is not sent for some reasons. In this case codec will be null
+/// * Different codecs are specified in the call endpoints
+class CodecMismatch implements QualityIssueEvent {
+  // Codec that is currently used or null if the video is not sent
+  final String? codec;
+
+  /// Issue level
+  final VIQualityIssueLevel level;
+
+  CodecMismatch({required this.level, required this.codec});
+}
+
+/// Represents class reporting that video resolution sent to the endpoint
+/// is lower than a captured video resolution. As a result it affects remote
+/// video quality on the remote participant side, but do not affect the quality
+/// of local video preview on the android application.
+///
+/// The issue level may vary during the call.
+///
+/// Possible reasons:
+///
+/// * High CPU load during the video call
+/// * Network issues such as poor internet connection or low bandwidth
+class LocalVideoDegradation implements QualityIssueEvent {
+  /// Sent frame size.
+  final FrameSize actualSize;
+
+  /// Captured frame size.
+  final FrameSize targetSize;
+
+  ///Issue level
+  final VIQualityIssueLevel level;
+
+  LocalVideoDegradation({
```

Comment:
I think it should be a private constructor, since it may be created only on the SDK side

Reply:
done, now quality issue classes having private named constructors